src/pdsfile/holdings_maintenance/pds4/pds4checksums.py

```
@@ -545,286 +542,48 @@ def update(pdsdir, selection=None, logger=None):
-            proceed = False
+def main():
+    result = _shelf_common.run_selection_main(SPEC, TASKS, sys.argv)
 
     # If everything went well, execute pdsinfoshelf too
-    if proceed and args.infoshelf:
+    if result.proceed and result.args.infoshelf:
         new_list = [a.replace('pdschecksums', 'pdsinfoshelf') for a in sys.argv]
         new_list = [a for a in new_list if a not in ('--infoshelf', '-i')]
-        result = subprocess.run(new_list)
-        sys.exit(result.returncode)
+        completed = subprocess.run(new_list, check=False)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)